es/2/interfaces.md

```diff
@@ -1,6 +1,77 @@
+
+Así que, para nuestro caso de uso, necesitamos crear una interfaz para el contrato de batalla pokemon en el contrato de entrenador. Esto permitirá que el contrato de entrenador se comunique con el contrato de batalla pokemon.
+
+## Usando Interfaces
+
+Las Interfaces se pueden añadir a los contratos, ya sea a través de definición en línea o importándolos de un archivo separado.
+
+La palabra `interface` se utiliza para definir una interfaz externa en línea:
+
+    interface Car:
+        def start() -> bool: nonpayable
+        def getSpeed() -> uint256: view
+
+La interfaz definida `Car` puede ser utilizada para hacer llamadas externas, dada una dirección de contrato:
+
+    @external
+    def Driver(some_address: address):
+        Car(some_address).start()
+
+El nombre de la interfaz (`Car`) también puede utilizarse como un tipo para las variables de almacenamiento. A continuación, se asigna un valor de dirección a la variable para acceder a esa interfaz:
+
+    car: Car
+
+    @external
+    def __init__(car_address: address):
+        self.car = Car(car_address)
+
+    @external
+    def getCarSpeed():
+        self.car.getSpeed()
+
+La especificación de la anotación `payable` o `nonpayable` indica que la llamada realizada al contrato externo podrá alterar el contrato de almacenamiento/datos, mientras que la `view`, la llamada `pure` NO podrá ser alterar el contrato de almacenamiento/datos. Además, `payable` permite enviar un valor distinto de cero junto con la llamada (lo que significa que se puede enviar ETH cuando se llama a una función payable).
+
+    interface Car:
+        def calculateDoubleSpeed() -> uint256: pure
+        def getSpeed() -> uint256: view
+        def increaseSpeed(): nonpayable
+        def payForPetrol(): payable
+
+    @external
+    def test(some_address: address):
+        Car(car_address).calculateDoubleSpeed()  # no puede cambiar almacenamiento
+        Car(car_address).getSpeed()  # no puede cambiar almacenamiento, pero se lee a sí mismo
+        Car(car_address).increaseSpeed()  # almacenamiento puede ser alterado
+        Car(car_address).payForPetrol(value=100)  # almacenamiento puede ser alterado, y el valor puede ser enviado
+
+> NOTA: `view` y la llamada `pure` usan un `STATICCALL` asegurando que no se pueda alterar el almacenamiento durante la ejecución.
+
+## Poniendolo a prueba
+
+Como el área de codificación sólo puede tener un archivo a la vez, hemos eliminado el contrato de batalla pokemon y añadido el contrato de entrenador. Puedes ver el contrato de batalla de pokemon [aquí](https://github.com/vyperfun/vyper.fun/blob/chapter1/assets/2/2.7-finished-code.vy).
```

Comment:
**Spanish**
```
Como el área de codificación sólo puede tener un archivo a la vez
```

**English (Generated)**
```
Since the encoding area can only have one file at a time
```

Here the word `encoding` should be `coding`. Please change the translation to reflect this  :smile:

Reply:
Translation is good, coding has both meanings in Spanish, however, I would prefer:

> Dado que el area donde codificamos nuestros contratos sólo puede tener un archivo a la vez...